lib/modem_info/modem_info_json.c

```diff
@@ -159,12 +168,76 @@ static int device_data_add(struct device_param *device, cJSON *json_obj)
 	total_len = json_add_data(&device->modem_fw, json_obj);
 	total_len += json_add_data(&device->battery, json_obj);
 	total_len += json_add_str(json_obj, "board", device->board);
-	total_len += json_add_str(json_obj, "appVer", device->app_version);
+	total_len += json_add_str(json_obj, "appVersion", device->app_version);
 	total_len += json_add_str(json_obj, "appName", device->app_name);
 
 	return total_len;
 }
 
+int modem_info_json_object_encode(struct modem_param_info *modem,
+				  cJSON *root_obj)
+{
+	int total_len = 0;
+	int ret;
+
+	if (root_obj == NULL || modem == NULL) {
+		return -EINVAL;
+	}
+
+	cJSON *network_obj	= cJSON_CreateObject();
+	cJSON *sim_obj		= cJSON_CreateObject();
+	cJSON *device_obj	= cJSON_CreateObject();
+
+	if (network_obj == NULL || sim_obj == NULL || device_obj == NULL) {
+		cJSON_Delete(network_obj);
+		cJSON_Delete(sim_obj);
+		cJSON_Delete(device_obj);
+		return -ENOMEM;
+	}
+
+	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_NETWORK)) {
+		ret = network_data_add(&modem->network, network_obj);
+
+		if (ret < 0) {
+			return ret;
```

Comment:
Memory leak here and below - you skip `network_obj` releasing. Either release objects manually before each `return`, or if you're not afraid of goto, you could use it for error handling; add a label at line 233, and jump from here. Just remeber to set the return value accordingly before jump (`total_len = ret`).

Reply:
Great input. I'll add the goto. Thanks for the information. :)